mm-tools/mboot_api.c

```diff
@@ -399,7 +419,23 @@ void mboot_upgrade(struct i2c_drv *iic)
 	HASH_ADD_STR(thread_pool, name, thr);
 }
 
-#define DRIVER_CHECK_FINISH(X) mboot_check(&X##_drv);
+#define DRIVER_REBOOT_MM(X) mm_reboot(&X##_drv);
+void mm_reboot(struct i2c_drv *iic)
+{
+    struct thread *thr = NULL;
+
+    thr = (struct thread *)malloc(sizeof(struct thread));
+    if (!thr) {
+        printf("%s-Failed to malloc struct thread!", iic->name);
+        exit(1);
+    }
+    strncpy(thr->name, iic->name, THR_NAME_LEN);
+
+    pthread_create(&thr->pth, NULL, reboot_op, (void *)iic);
+    HASH_ADD_STR(thread_pool, name, thr);
+}
+
+#define DRIVER_CHECK_FINISH(X) mboot_check(&X##_drv); // Function name change is suggested
 void mboot_check(struct i2c_drv *iic)
```

Comment:
@AmamiyaRinyuki Thanks, so mboot_finish or mboot_done can be used, how about it ?